reactor-core/src/test/java/reactor/core/publisher/scenarios/MonoTests.java

```diff
@@ -304,4 +314,156 @@ public void monoCacheContextHistory() {
 		Assertions.assertThat(cacheHit3).as("cacheHit3").isEqualTo("GOOD1");
 		Assertions.assertThat(contextFillCount).as("cacheHit3").hasValue(4);
 	}
+
+	@Test
+	public void monoFromMonoDoesntCallAssemblyHook() {
+		final Mono<Integer> source = Mono.just(1);
```

Comment:
May I ask why not parameterized? :)

Reply:
didn't go through the trouble, as splitting the test was enough vs changing the test runner and dealing with the boilerplate of providing params (plus with independent tests the name can reflect the style of source, not the concrete name of the operator used)